kolibri/core/content/views.py

```diff
@@ -131,14 +137,33 @@ def parse_html(content):
                 )
             },
         )
-        return html5lib.serialize(
+
+        # Currently, html5lib strips the doctype, but it's important for correct rendering, so check the original
+        # content for the doctype and, if found, prepend it to the content serialized by html5lib
+        doctype = None
+        try:
+            encoding = p.documentEncoding
+            if not encoding:
+                encoding = "utf-8"
+            doctype = re.match("(?i)<!DOCTYPE[^<>]*(?:<!ENTITY[^<>]*>[^<>]*)?>", content.decode(encoding))
+        except:
+            # Losing the doctype could lead to some rendering issues, but they are usually not severe enough
+            # to be worth stopping the content from loading entirely.
+            pass
```

Comment:
let's make sure to only swallow the specific exception your expecting here. Might also be worth logging a warning so it's not completely silent, in case we need to debug in the future

Reply:
Definitely will add logging. The most likely error I'm expecting would be a unicode decode error, but my priority here is that even an unexpected failure to detect doctype should not cause a regression in the ability to load content. (Consider that Kolibri currently is stripping the doctype from all HTML5 apps.)

Reply:
Okay makes sense. in that case you can add `# noqa: E722` on line 149 to ignore the linting warning